### PR TITLE
docs: add more insights about Uno's code style/conventions

### PR DESCRIPTION
**Pull request by ghuntley**
## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## Other information

Added insights from https://github.com/unoplatform/uno/issues/1834#issuecomment-542161088

### CODE REVIEW

**Comment by ghuntley** [comment]
@jeromelaban can you please ship some guidance into this branch (or reply back) about `SampleControlInfo` naming conventions? How should `SampleControlInfoAttribute` be used and what is a category?

There seems to be multiple conventions in use within the samples app:

image

# example

Assuming we are creating samples for the `Windows.UI.Xaml.Controls.ContentDialog` control which is right?

```
[SampleControlInfo("ContentDialog", "ContentDialog_Closing", description: "Tests for ContentDialog.Closing event")]
```

Or

```
[SampleControlInfo("Windows.UI.Xaml.Controls", "ContentDialog_Closing", description: "Tests for ContentDialog.Closing event")]
``

Or ... something else

# remarks

The signature for `SampleControlInfoAttribute` currently is as follows:

```csharp
public SampleControlInfoAttribute(string category, string controlName, Type viewModelType = null, bool ignoreInAutomatedTests = false, string description = null)
{
   ...
}
```

Changing the signature for `SampleControlInfoAttribute` to be this would be an improvement:


```csharp
public SampleControlInfoAttribute(string namespace, string controlName, string testScenario, Type viewModelType = null, bool ignoreInAutomatedTests = false, string description = null)
{
   ...
}
```

**Comment by ghuntley** [comment]
@jeromelaban additionally can you please do the same with naming of folders and tests

image

**Comment by ghuntley** [comment]
There is some content at https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md#creating-ui-tests which can be used as a baseline.

**Comment by davidjohnoliver** [comment]
- Using namespaces as a category mainly makes sense for non-UI features. Sticking every control in the same category would be a bad idea. 
 - That said we should standardise on one convention - eg 'ContentDialog' over 'ContentDialogTests'. (It's a mess for mainly historical reasons.)
 - for the *folder name* however I prefer `/FooTests` over just `Foo`, simply because if the namespace is just `...Foo`, then trying to create `new Foo()` in code-behind (in *any* sample) gives rise to compiler complaints.

**Comment by jeromelaban** [comment]
`ContentDialogTests` must be used to avoid automatic naming, which conflicts with the original control's name.

File: doc/articles/contributing/guidelines/code-style.md [hunk 1]
```diff
@@ -32,19 +39,45 @@ if (condition)
...
 ## Integration Tests
 
 ```csharp
-[ActivePlatforms(Platform.Android, Platform.Browser, Platform.iOS)]
 [TestFixture]
 public class LocalSettings_Tests : SampleControlUITestBase
 {
     [Test]
     [AutoRetry]
+    [ActivePlatforms(Platform.Android, Platform.Browser, Platform.iOS)]
```
Comment (jeromelaban) [inline doc/articles/contributing/guidelines/code-style.md]: This must not be the default. Tests should run on all platforms by default, even those not listed yet.

**Comment by carldebilly** [comment]
Since the merging of the "clipping" PR, there's a new feature where the sample name is not required on the attribute anymore: it will use the class name instead.

Same for sample category: it will use the namespace name (local name, not full name)

**Comment by jeromelaban** [comment]
Many of these items have been documented in more recent PRs.

**Pull request closed without merging**